go/exosocket/index.go

```diff
@@ -0,0 +1,106 @@
+package exosocket
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Originate/exocom/go/structs"
+	"github.com/Originate/exocom/go/utils"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+)
+
+// ExoSocket is the low level API to Exocom
+type ExoSocket struct {
+	config                      Config
+	shouldReconnectOnDisconnect bool
+	socket                      *websocket.Conn
+	messageChannel              chan structs.Message
+}
+
+// NewExoSocket returns an ExoSocket with the given config
+func NewExoSocket(config Config) *ExoSocket {
+	return &ExoSocket{config: config}
```

Comment:
This is a new function as the object should not be instantiated directly

Reply:
Sounds good. I'll look into that. The interface would probably just include the public methods as the private fields are not used outside this file.